ee/tabby-ui/app/files/components/code-editor-view.tsx

```diff
@@ -65,8 +102,46 @@ const CodeEditorView: React.FC<CodeEditorViewProps> = ({ value, language }) => {
         highlightTagExtension(tags)
       )
     }
+
     return result
-  }, [value, tags, language, editorRef.current])
+  }, [value, tags, language])
+
+  React.useEffect(() => {
+    const onClickLineMenu = (data: LineMenuActionEventPayload) => {
+      if (!line) return
+      if (data.action === 'copy_permalink') {
+        copyToClipboard(window.location.href)
+        return
+      }
+      if (data.action === 'copy_line') {
+        const lineNumber = parseInt(line)
+        const lineObject = editorView?.state?.doc?.line(lineNumber)
+        if (lineObject) {
+          copyToClipboard(lineObject.text)
+        }
+      }
+    }
+    emitter.on('line_menu_action', onClickLineMenu)
+
+    return () => {
+      emitter.off('line_menu_action', onClickLineMenu)
+    }
+  }, [value, line])
+
+  React.useEffect(() => {
+    if (initialized.current) return
+    if (line && editorView && value) {
+      console.log('=====in 2')
```

Comment:
Seems like some testing code and comments need to be removed

Reply:
other LGTM, tested in my local as well

Reply:
removed